### ISSUE TITLE
Would like to be able to execute interactive command

### ISSUE THREAD

**Issue by hpe-ykoehler** [devcontainers/cli#59]
In my env, developer will likely want to run devcontainer exec bash and then be able to navigate the container and do stuff within using command-line interface.

As such, I think the option to exec the command in interactive mode is important.

**Comment by hpe-ykoehler** [comment]
When I use the devcontainer exec of today to run "bash" I cannot type anything, the output shows the prompt and the container is not accepting inputs.

That is normally because when using docker, to run bash we must use -ti to make the docker knows we require a TTY and tha that the session will be interactive so that it connects the stdin to the container (or someting along those line).

So the test is "devcontainer exec bash" and see that you get a functional terminal, which in my setup I do not get.

**Comment by Nking92** [comment]
+1 to this request. There are devs I work with who don't use VSCode as their editor but would still find devcontainers useful in their workflow. An interactive shell is essential for them.

In the meantime, it's possible to open a terminal session by using `docker container ls` to get the running container ID, then run bash with `docker run -it <id> /bin/bash`


```
nk@DESKTOP-LIH0TIU:~/code/vscode-remote-try-rust$ docker container ls
CONTAINER ID   IMAGE                                                             COMMAND                  CREATED         STATUS             PORTS     NAMES
d5b67fea7ce8   vsc-vscode-remote-try-rust-d007786a17c62e0d1a94f4da56714892-uid   "/bin/sh -c 'echo Co…"   5 minutes ago   Up 5 minutes                 kind_brahmagupta

nk@DESKTOP-LIH0TIU:~/code/vscode-remote-try-rust$ docker exec -it d5b67fea7ce8 /bin/bash
root ➜ / $
```

**Comment by zflat** [comment]
This works OK from within a workspace directory:

```bash
docker exec -it -u $(id -u) $(devcontainer up --workspace-folder . | jq -r .containerId) bash
```
It will get the ID of a container that was previously started from the [Remote - Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) VSCode extension.

**Comment by mscottford** [comment]
> This works OK from within a workspace directory:
> 
> ```shell
> docker exec -it -u $(id -u) $(devcontainer up --workspace-folder . | jq -r .containerId) bash
> ```
> 
> It will get the ID of a container that was previously started from the [Remote - Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) VSCode extension.

Thanks for this tip. I had to remove the `-u $(id -u)` parameter to get this to work. Otherwise, I get output that looks like this:

```
❯ docker exec -it -u $(id -u) $(devcontainer up --workspace-folder . | jq -r .containerId) bash
[9 ms] @devcontainers/cli 0.9.1.
whoami: cannot find name for user ID 501
rm: cannot remove '/usr/local/share/nvm/current': Permission denied
I have no name!@fe0ad01686b5:/$
```

**Comment by behrica** [comment]
I would as well add that to run a bash in a devcontainer is the "standard"  task, which non-vscode-users will do with a devcontainer.
So supporting an `-i` flag in `devcontainer exec` would be very useful.

**Comment by mickeahlinder** [comment]
I've tried to debug this for a couple of hours today, reversing the CLI code.
I've come to the conclusion that the CLI does indeed initiate `docker exec`-session using `-ti` as arguments, but does not read input coming from the cli-host, in other words: The user types a command and presses Enter. However, when hijacking the PTY stream from `node-pty` and manually sending a command to the process spawned (`p.write('ls\r')`) works and prints the output.
So the input from the cliHost is "blocked" somewhere.
I would gladly look into this further, but I lack the technical knowledge of how the streams fit together. With some assistance or just some pointing as to where the link might be "broken" I might be able to continue.

**Comment by chrmarti** [comment]
I'm looking into changing `devcontainer exec` to have stdout, stderr and exit code mirror those of the command run inside the dev container and also have stdin connected.

Is anyone relying on the fact that devcontainer exec currently returns a short JSON object with the “outcome” property on stdout? (We are aware that our CI action does, but that we can easily update.)

**Comment by chrmarti** [comment]
Available with version 0.33.0 of the CLI.

**Comment by hpe-ykoehler** [comment]
Thanks !
